<launch>

    <param name="robot_description" textfile="$(env ROBOTOLOGY_ROOT)/robots/iit-cogimon-ros-pkg/cogimon_urdf/urdf/cogimon.urdf"/>
    <param name="robot_description_semantic" textfile="$(env ROBOTOLOGY_ROOT)/robots/iit-cogimon-ros-pkg/cogimon_srdf/srdf/cogimon.srdf"/>
    <param name="cartesian/problem_description" textfile="$(env ROBOTOLOGY_ROOT)/external/limit_cycle_walking/configs/cartesio_stack.yaml"/>

    
    <!-- param name="cartesian/com/max_velocity_linear" type="double" value="100.0"/ -->
    <!-- param name="cartesian/com/max_acceleration_linear" type="double" value="100.0"/ -->
    
    <!-- include file="$(find cartesian_interface)/launch/cartesio.launch"/ -->
    
    <arg name="solver" default="OpenSot"/>
    <arg name="prefix" default=""/>
    <arg name="use_xbot_config" default="false"/>
    <arg name="rate" default="100.0"/>
    <arg name="tf_prefix" default="ci"/>
    <arg name="markers" default="true"/>
    <arg name="namespace" default="/"/> 
    <arg name="is_model_floating_base" default="true"/>
    <arg name="xbot" default="false"/>
       
    <group ns="$(arg namespace)">
    
        <remap if="$(arg xbot)" from="robot_description" to="xbotcore/robot_description"/>
    
        <node pkg="cartesian_interface" type="ros_server_node" 
                                        name="ros_server_node" 
                                        required="true" 
                                        output="screen" 
                                        launch-prefix="$(arg prefix)"
                                        rosmon-stop-timeout="1000">
                                        
            <param name="is_model_floating_base" value="$(arg is_model_floating_base)"/>
            <param name="model_type" value="RBDL"/>
            <param name="solver" value="$(arg solver)"/>
            <param name="use_xbot_config" value="$(arg use_xbot_config)"/>
            <param name="rate" value="$(arg rate)"/>
            <param name="tf_prefix" value="$(arg tf_prefix)"/>
            
        </node>

        <node if="$(arg markers)" pkg="cartesian_interface" type="marker_spawner" name="interactive_markers" output="screen">
            <param name="tf_prefix" value="$(arg tf_prefix)"/>
        </node>

    </group>

</launch>
